<html>
<body>
<code>finally</code> 블록 내에서 실행되지 않은 JUnit 3의 <code>super.tearDown()</code> 메서드 호출을 보고합니다.
<code>super.tearDown()</code> 을 호출하기 전에 예외가 발생할 수 있는 <code>tearDown()</code> 메서드 내에서 다른 메서드를 호출하는 경우 불일치와 누수가 발생할 수 있습니다.
<p><b>예:</b></p>
<pre><code>
  <b>public class</b> AnotherTest <b>extends</b> CompanyTestCase {
    <b>private</b> Path path;

    @Override
    <b>protected void</b> setUp() <b>throws</b> Exception {
      <b>super</b>.setUp();
      path = Files.createTempFile("abcde", ".tmp");
    }

    @Override
    <b>protected void</b> tearDown() <b>throws</b> Exception {
      Files.delete(path);
      <b>super</b>.tearDown();
    }
  }
</code></pre>
<p>개선된 코드:</p>
<pre><code>
  <b>public class</b> AnotherTest <b>extends</b> CompanyTestCase {
    <b>private</b> Path path;

    @Override
    <b>protected void</b> setUp() <b>throws</b> Exception {
      <b>super</b>.setUp();
      path = Files.createTempFile("abcde", ".tmp");
    }

    @Override
    <b>protected void</b> tearDown() <b>throws</b> Exception {
      <b>try</b> {
        Files.delete(path);
      } <b>finally</b> {
        <b>super</b>.tearDown();
      }
    }
  }
</code></pre>
  <!-- tooltip end -->
</body>
</html>